packages/ng/tooltip/trigger/tooltip-trigger.directive.ts

```diff
@@ -1,156 +1,339 @@
+		const ref = this.overlayRef.attach(portal);
+		if (this.luTooltip) {
+			ref.instance.content = this.luTooltip;
+		} else if (this.luTooltipWhenEllipsis) {
+			ref.instance.content = this.#host.nativeElement.innerText;
 		}
+		ref.instance.id = this.ariaDescribedBy;
+		// On tooltip leave => trigger close
+		ref.instance.mouseLeave$.pipe(takeUntilDestroyed(ref.instance.destroyRef)).subscribe(() => this.close$.next());
+		// On tooltip enter => trigger open to keep it opened
+		ref.instance.mouseEnter$.pipe(takeUntilDestroyed(ref.instance.destroyRef)).subscribe(() => this.open$.next());
 	}
 
-	ngAfterViewInit() {
-		this._checkTarget();
-	}
-	ngOnDestroy() {
-		this._cleanUpSubscriptions();
-		if (this._popoverOpen) {
-			this.closePopover();
+	private closeTooltip(): void {
+		if (this.overlayRef) {
+			this.overlayRef.detach();
+			delete this.overlayRef;
 		}
-		this.destroyPopover();
-	}
-	protected _emitOpen(): void {
-		this.onOpen.emit();
-	}
-	protected _emitClose(): void {
-		this.onClose.emit();
 	}
 
-	protected override _createOverlay(): OverlayRef {
-		if (!this._overlayRef) {
-			this._portal = new ComponentPortal(LuTooltipPanelComponent, this._viewContainerRef);
-			const config = this._getOverlayConfig();
-			this._subscribeToPositions(config.positionStrategy as FlexibleConnectedPositionStrategy);
-			this._overlayRef = this._overlay.create(config);
+	private setAccessibilityProperties(tabindex: number | null): void {
+		if (this.#disabled || (this.luTooltipWhenEllipsis && !this.hasEllipsis())) {
+			this.#renderer.removeAttribute(this.#host.nativeElement, 'tabindex');
+			return;
 		}
 
-		return this._overlayRef;
+		const tag = this.#host.nativeElement.tagName.toLowerCase();
+		const nativelyFocusableTags = ['a', 'button', 'input', 'select', 'textarea'];
+		const isNatevelyFocusableTag = nativelyFocusableTags.includes(tag);
+
+		const hasATabIndex = this.#host.nativeElement.getAttribute('tabindex') !== null;
+
+		if (!isNatevelyFocusableTag && !hasATabIndex) {
+			this.#renderer.setAttribute(this.#host.nativeElement, 'tabindex', tabindex.toString());
+		}
 	}
 
-	protected override _attachPortalToOverlay(): void {
-		const componentRef = this._overlayRef.attach(this._portal);
-		this._panel = componentRef.instance;
-		this._panel.content = this._tooltipContent;
+	/**
+	 * Hacky af but let's explain everything
+	 * This method checks for ellipsis by cloning the node and checking its width against original element.
+	 *
+	 * We used to do this using scrollWidth but the thing is, it's a rounded value. Sometimes,
+	 * you'd get true while it should be false and vice-versa, because of rounding.
+	 *
+	 * We could also use getBoundingClientRect() but it only considers text content, meaning that if ellipsis is caused by
+	 * any margin or padding, it won't be detected
+	 *
+	 * @private
+	 */
+	private hasEllipsis(): boolean {
+		if (window.getComputedStyle(this.#host.nativeElement).textOverflow !== 'ellipsis') {
+			return false;
```

Comment:
should we log a warning if the user ask for `whenEllispsis` but `text-overflow` is not correct?

Reply:
We could but that would possibly cause a lot of warnings in various places since this is not a new component, it's a refactor of an existing one.